caikit/runtime/service_generation/signature_parsing/docstrings.py

```diff
@@ -68,10 +69,11 @@ def is_optional(fn: Callable, arg_name: str) -> bool:
         arg_name: The name of the parameter that we should try to get the type of
             e.g. "raw_document"
     """
-    docstring = docstring_parser.parse(fn.__doc__)
-    if not docstring:
+    try:
+        docstring = docstring_parser.parse(fn.__doc__)
+    except docstring_parser.ParseError:
         log.warning("Failed to parse the docstring for %s", fn.__name__)
-        return False
+        return None
```

Comment:
nit: this function returns a `bool`, should this still be `False`? Or is there a semantic difference between `False` and `None` that we're checking for later?

Reply:
I considered throwing an exception here and let the caller of this function handle that... I could do that